src/caffe/syncedmem.cpp

```diff
@@ -3,26 +3,41 @@
 #include "caffe/util/math_functions.hpp"
 
 namespace caffe {
+SyncedMemory::SyncedMemory()
```

Comment:
All the extra debugging code is a bit ugly, but if it makes debugging easier and doesn't affect performance outside of debug mode (and it looks like it shouldn't as everything's in `#ifdef DEBUG` guards), I'm ok with it.  `check_device` could use a comment though.